File: src/lib/pricing-db.ts
import { eq } from "drizzle-orm";
import { db } from "./db";
import { pricing } from "./schema";
import type { PricingRow, PriceUnit } from "./pricing";

/** DB-backed pricing access (kept separate so pricing.ts stays client-safe). */

export async function readPricing(): Promise<PricingRow[]> {
  const rows = await db.select().from(pricing);
  return rows.map((r) => ({
    model: r.model,
    unitCostCents: r.unitCostCents,
    unit: r.unit as PriceUnit,
    notes: r.notes,
  }));
}

export async function updatePricing(
  model: string,
  unitCostCents: number,
  unit: PriceUnit
): Promise<void> {
  await db
    .insert(pricing)
    .values({ model, unitCostCents, unit })
    .onConflictDoUpdate({
      target: pricing.model,
      set: { unitCostCents, unit },
    });
}

File: backend/apps/generation/pricing_db.py
"""Port of src/lib/pricing-db.js — DB-backed pricing access."""

from .models import Pricing


def read_pricing() -> list[dict]:
    return [
        {"model": p.model, "unitCostCents": p.unit_cost_cents, "unit": p.unit, "notes": p.notes}
        for p in Pricing.objects.all()
    ]


def update_pricing(model: str, unit_cost_cents: int, unit: str) -> None:
    Pricing.objects.update_or_create(model=model, defaults={"unit_cost_cents": unit_cost_cents, "unit": unit})
